### PR TITLE
feat(cart): Add ShoppingCartPage, enhance cart interactions, item adjustments, and introduce order summary

### PR DESCRIPTION
**Pull request by irtassedat**
Description:
This PR introduces several enhancements to the shopping cart functionality and integrates an order summary feature into the shopping cart page. Below are the key changes and their impacts on the project:

Features Implemented:
Dynamic Shopping Cart Page: Added a new ShoppingCartPage component to dynamically display products added to the cart along with their quantities.
Product Quantity Adjustment: Enabled users to increase or decrease product quantities directly from the shopping cart, improving the shopping experience.
Order Summary Integration: Introduced an order summary box on the right side of the shopping cart page, providing users with a clear overview of the total price, shipping costs, discounts, and the grand total.
Selectable Products for Order Creation: Implemented functionality allowing users to select or deselect products, enabling them to choose which products they want to include in their order.
Technical Details:
Utilized React state and Redux for state management across the shopping cart and order summary features.
Added new actions and reducers to handle product selection and quantity adjustments within the cart.
How to Test:
Add multiple products to the shopping cart and observe the dynamic updates to the product list and order summary.
Increase and decrease product quantities and verify the correct calculation of the total price.
Select and deselect products and ensure that only selected products are considered in the order summary.

Additional Notes:
This PR addresses tasks 14, 15, and 16 related to shopping cart enhancements and order summary integration.

### CODE REVIEW

File: src/components/LayoutComponents/Header.jsx
Comment (gokhanozdemir) [inline src/components/LayoutComponents/Header.jsx:1]: useState ve useEffect hook'larını kullanırken bağımlılık dizilerini doğru şekilde yönettiğinden emin ol. Bir değişkenin state'i değiştiğinde gereksiz re-render'ları önlemek için kontrol et.

File: src/pages/ShoppingCartPage.jsx
Comment (gokhanozdemir) [inline src/pages/ShoppingCartPage.jsx:1]: Sepetteki ürünlerin adetini artırma veya azaltma işlevselliğinde, kullanıcının sıfırdan az bir adet seçmesini engelleyecek bir kontrol mekanizması ekle. Bu, olası hataları önler.

**Review by gokhanozdemir — COMMENTED** [review]
Hatalı bir yaklaşım görmedim, iyileştirmeleri inline ekledim. 🚀 Eline sağlık.

File: src/pages/ShoppingCartPage.jsx
Comment (irtassedat) [reply src/pages/ShoppingCartPage.jsx:1]: @gokhanozdemir decrementProductCount actionuna ekledim bu mekanizmayı, ürünleri kaldır butonum daha anlamlı oldu böylelikle

File: src/components/LayoutComponents/Header.jsx
Comment (irtassedat) [reply src/components/LayoutComponents/Header.jsx:1]: ilgili yapımda [user] yerine [] döndürdüğüm zaman sayfama refresh isteği atmadan verify çalışmadan userbilgileri render olmuyor nasıl bir iyileştirmeden söz ediyoruz tam anlayamadım @gokhanozdemir

**Pull request merged**